plugins/Terminal/js/utils/helpers.js

```diff
@@ -0,0 +1,260 @@
+			closed = true
+		}
+	}
+
+	siac.on('error', () => {
+		console.log('\tPROGRAM ERRORED')
+		streamClosed()
+	})
+	siac.on('close', () => {
+		console.log('\tPROGRAM CLOSED')
+		streamClosed()
+	})
+
+	//If window is small auto close command overview so we can see the return value.
+	if (document.getElementsByClassName('command-history-list')[0].offsetHeight < 180) {
+		actions.hideCommandOverview()
+	}
+
+	return siac
+}
+
+export const httpCommand = function(commandStr, actions, newid) {
+	const commandString = commandStr
+	const originalCommand = commandStr.replace(/\s*\s/g, ' ').trim()
+
+	//Remove surrounding whitespace and leading siac command.
+	if (commandString.startsWith('siac')) {
+		commandString = commandString.slice(4).trim()
+	} else if (commandString.startsWith('./siac')) {
+		commandString = commandString.slice(6).trim()
+	}
+
+	//Parse arguments.
+	const args = commandString.split(' ')
+
+	//Add address flag to siac.
+	let siaAddr = url.parse('http://localhost:9980')
+
+	if (args.indexOf('-a') === -1 && args.indexOf('--address') === -1) {
+		if (SiaAPI.config.attr('address')) {
+			//Load default address.
+			siaAddr = url.parse('http://' + SiaAPI.config.attr('address'))
+		}
+	} else {
+		//Parse address flag.
+		let index = args.indexOf('-a')
+		if  (index === -1) {
+			index = args.indexOf('--address')
+
+		}
+		if (index < args.length-1) {
+			siaAddr = url.parse('http://' + args[index+1])
+		}
+		args.splice(index, 2)
+		commandString = args.join(' ')
+	}
+
+	let apiURL = ''
+	switch (commandString) {
+	case 'wallet unlock':
+		apiURL = '/wallet/unlock'
+		break
+
+	case 'wallet load seed':
+		apiURL = '/wallet/seed'
+		break
+
+	default:
+		console.log(`ERROR Command is not an http command. Rerouting. ${commandString}`)
+		return spawnCommand(commandString, actions).stdin
+	}
+
+	//Spawn new command if we are good to go.
+	const newCommand = Map({ command: originalCommand, result: '', id: newid, stat: 'running' })
+	actions.addCommand(newCommand)
+
+	//Update the UI when the process receives new ouput.
+	const consumeChunk = function(chunk) {
+		console.log('Data chunk ' + chunk)
+		let newChunk = chunk
+		if (chunk.toString().trim() === '{"Success":true}') {
+			newChunk = 'Success'
+		}
+		actions.updateCommand(newCommand.get('command'), newCommand.get('id'), newChunk)
+	}
+
+	let closed = false
+	const streamClosed = function() {
+		if (!closed) {
+			actions.endCommand(newCommand.get('command'), newCommand.get('id'))
+			closed = true
+		}
+	}
+
+	//If window is small auto close command overview so we can see the return value.
+	if (document.getElementsByClassName('command-history-list')[0].offsetHeight < 180) {
+		actions.hideCommandOverview()
+	}
+
+	const options = {
+		hostname: siaAddr.hostname,
+		port: siaAddr.port,
+		path: apiURL,
+		method: 'POST',
+		headers: {
+			'User-Agent': 'Sia-Agent',
+			'Content-Type': 'application/x-www-form-urlencoded',
+		},
+	}
+	const req = http.request(options, (res) => {
```

Comment:
@lukechampine, @johnathanhowell, just wanted to confirm that I should remove the http request for the SiadAPI.call. The current http call can handle `--address` attributes and will auto link to the running siad instance otherwise. I'm not sure what we gain by using the SiadAPI, but it would be simple drop in.